src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler.properties

```diff
@@ -512,6 +512,9 @@ compiler.err.duplicate.unconditional.pattern=\
 compiler.err.unconditional.pattern.and.default=\
     switch has both an unconditional pattern and a default label
 
+compiler.err.guard.has.constant.expression.false=\
+    this case label has a when constant expression that is false
```

Comment:
`when` here should be: `''when''`

Reply:
True. I improved the error message a bit.